node/packages/parser/src/ethereum/uniV2.ts

```diff
@@ -37,9 +38,15 @@ export class Parser {
     }[this.network]
   }
 
+  isSupportedTransaction(tx: Tx) {
+    const isUniV2Contract = txMatchesContract(tx, ROUTER_CONTRACT)
+    const hasNoConfirmations = tx.confirmations === 0
+    return hasNoConfirmations && isUniV2Contract
+  }
+
   async parse(tx: Tx): Promise<ParseTxSpecific | undefined> {
-    if (!tx.ethereumSpecific?.data) return
-    if (tx.confirmations !== 0) return
+    if (!this.isSupportedTransaction(tx)) return
+    if (!(isEthereumSpecific(tx.ethereumSpecific) && tx.ethereumSpecific.data)) return
```

Comment:
can just move this into the parsable/supported check to keep all of that logic together

Reply:
Unfortunately, the compiler isn't smart enough to realise that `tx.ethereumSpecific.data` is defined if we do the check inside `isParsable`, so we then need to do a bunch of checks in `parse()`.

I'm not aware of a way around this - am I missing something?

Reply:
No you are correct, we do lose the type narrowing scope by moving that into `isParsable`. Avoiding additional undefined checks is the right call.

Reply:
The only other option would be just forgoing the `isParsable` function and just having the few checks at the start of the `parse` function.

Reply:
Yeah, the checks probably aren't even worth splitting into `isParsable()` - I've removed all instances of it for now in https://github.com/shapeshift/unchained/pull/238/commits/68fc4e8bdfc87f4198611e671ec694da193f39d4.